lib/apis/moray.js

```diff
@@ -493,6 +488,34 @@ Moray.prototype._createSearch = function (params, cb) {
 };
 
 
+/**
+ * List all VMs for a given server from the VM bucket.
+ * @param  {string}     uuid uuid of the server whos vms should be fetched.
+ * @param  {Function}   cb  function of the form function(err, vms) {...}
+ */
+Moray.prototype.listVmsForServer = function listVmsForServer(uuid, cb) {
```

Comment:
The biggest remaining question I have is whether we can change Moray.prototype.listVmsForServer such that we don't grab VMs which have zone_state=destroyed && state=destroyed. The reason for this is that as far as I understand it, the current code will load all VMs from the beginning of history. Which, especially for CNs handling docker VMs, could be a very large amount. We'll likely hit limits on that and need to do paging and this will just get slower and slower over time. If we ignore the destroyed && destroyed, we would only be looking at the current set of active VMs and those being destroyed. 

It seems like this might be possible since validStateChange() only checks for zone_state !== 'destroyed' which seems like it will be fine?